packages/shopify-app-remix/src/auth/helpers/ensure-cors-headers.ts

```diff
@@ -8,6 +8,7 @@ export interface EnsureCORSFunction {
 export function ensureCORSHeadersFactory(
   params: BasicParams,
   request: Request,
+  extraHeaders: string[] = [],
```

Comment:
Unfortunately, the extension may add arbitrary headers to the request (it's just a plain `fetch` call after all, see [line 28 in this example](https://shopify.dev/docs/apps/checkout/product-offers/post-purchase/getting-started#step-3-build-the-extension-ui)), and every header in the request needs to be cleared, so I think the optional header is justified and the easiest instruction is "add every header you're sending to the list", since the only default one is `Authorization` because of the token

Reply:
Let's add `"Content-Type": "application/json"` by default, and still support the additional headers